File: app/src/main/java/com/uvnode/typhoon/extensions/source/Rankable.java
package com.uvnode.typhoon.extensions.source;

import com.uvnode.typhoon.extensions.api.ApiCallbacks;
import com.uvnode.typhoon.extensions.model.Ranking;

public interface Rankable {

    void getRankings(ApiCallbacks callbacks) throws Exception;

    void getSeriesList(ApiCallbacks callbacks, Ranking ranking) throws Exception;
}

File: app/src/main/java/com/uvnode/typhoon/extensions/source/Rankable.kt
package com.uvnode.typhoon.extensions.source

import com.uvnode.typhoon.extensions.api.ApiCallbacks
import com.uvnode.typhoon.extensions.model.Ranking

interface Rankable {
    @Throws(Exception::class)
    fun getRankings(callbacks: ApiCallbacks?)

    @Throws(Exception::class)
    fun getSeriesList(callbacks: ApiCallbacks, ranking: Ranking)
}